<a href="https://colab.research.google.com/github/Leo270499/Apuntes/blob/main/Dataset_S%C3%A1nchez.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Primera Entrega | Data Science

### ***1. Cargar las librerías***

In [2]:
%config IPCompleter.greedy=True
import pandas as pd
import numpy as np

### ***2. Cargar la base de datos***

In [4]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [5]:
from google.colab import files

In [6]:
from google.colab import drive

drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


#### ***2.1 Primera base de datos***
#### Perth House Price
> Perth es una ciudad de Australia conocida por su prospera economía, hermosos paisajes y alta calidad de vida. A través de esta base de datos, buscamos predecir el precio de las propiedades de esta ciudad.

*   La ubicación.
*   El tamaño de la propiedad.
*   Las comodidades cercanas.
*   El número de habitaciones, etc.

> Con algoritmos de ciencia de datos, interé explicar y predecir el precio de las propiedades de esta ciudad.



In [7]:
import pandas as pd
df1= pd.read_csv("/content/drive/MyDrive/Primera Entrega (Data Science)/Perth Data.csv")
df1.shape

(33656, 19)

Shape: Esta base de datos...

*   Tiene 33656 filas y 19 columnas.
*   Tiene 33656 registros y 19 variables.

In [8]:
df1.head()

ADDRESS      SUBURB   PRICE  BEDROOMS  BATHROOMS  GARAGE  \
0    1 Acorn Place  South Lake  565000         4          2     2.0   
1      1 Addis Way       Wandi  365000         3          2     2.0   
2  1 Ainsley Court     Camillo  287000         3          1     1.0   
3  1 Albert Street    Bellevue  255000         2          1     2.0   
4     1 Aman Place   Lockridge  325000         4          1     2.0   

   LAND_AREA  FLOOR_AREA  BUILD_YEAR  CBD_DIST               NEAREST_STN  \
0        600         160      2003.0     18300  Cockburn Central Station   
1        351         139      2013.0     26900           Kwinana Station   
2        719          86      1979.0     22600           Challis Station   
3        651          59      1953.0     17900           Midland Station   
4        466         131      1998.0     11200        Bassendean Station   

   NEAREST_STN_DIST  DATE_SOLD  POSTCODE   LATITUDE   LONGITUDE  \
0              1800  09-2018\r      6164 -32.115900  115.842450   
1              4900  02-2019\r      6167 -32.193470  115.859554   
2              1900  06-2015\r      6111 -32.120578  115.993579   
3              3600  07-2018\r      6056 -31.900547  116.038009   
4              2000  11-2016\r      6054 -31.885790  115.947780   

                    NEAREST_SCH  NEAREST_SCH_DIST  NEAREST_SCH_RANK  
0   LAKELAND SENIOR HIGH SCHOOL          0.828339               NaN  
1                ATWELL COLLEGE          5.524324             129.0  
2  KELMSCOTT SENIOR HIGH SCHOOL          1.649178             113.0  
3  SWAN VIEW SENIOR HIGH SCHOOL          1.571401               NaN  
4                 KIARA COLLEGE          1.514922               NaN

In [9]:
df1.columns

Index(['ADDRESS', 'SUBURB', 'PRICE', 'BEDROOMS', 'BATHROOMS', 'GARAGE',
       'LAND_AREA', 'FLOOR_AREA', 'BUILD_YEAR', 'CBD_DIST', 'NEAREST_STN',
       'NEAREST_STN_DIST', 'DATE_SOLD', 'POSTCODE', 'LATITUDE', 'LONGITUDE',
       'NEAREST_SCH', 'NEAREST_SCH_DIST', 'NEAREST_SCH_RANK'],
      dtype='object')

##### *Descripción de las variables*

*   `ADDRESS`: Dirección de la propiedad.
*   `SUBURB`: Suburbios donde pertenece la propiedad.
*   `PRICE`: Precio al que se vendio la propiedad.
*   `BEDROOMS`: Número de habitaciones.
*   `BATHROOMS`: Número de baños.
*   `GARAGE`: Número de garajes.
*   `LAND_AREA`: Superficie del terrerno. (m^2)
*   `FLOOR_AREA`: Superficie interna de la propiedad. (m^2)
*   `BUILD_YEAR`: Año que se construyo la propiedad.
*   `CBD_DIST`: Distancia desde el centro de Perth.
*   `NEAREST_STN`: Estación de transporte público más cercana.
*   `NEAREST_STN_DIST`: La distancia a la estación más cercana (m).
*   `DATE_SOLD`: Fecha que se vendió la propiedad.
*   `POSTCODE`: Codigo Postal
*   `LATITUDE`: Latitud
*   `LONGITUDE`: Longitud
*   `NEAREST_SCH`: Escuela más cercana.
*   `NEAREST_SCH_DIST`:Distancia a la escuela más cercana.
*   `NEAREST_SCH_RANK`: Clasificación de la escuela más cercana.










#### ***2.2 Segunda base de datos***
#### Credit Card Fraud Detection
> Esta base de datos contiene las transacciones realizadas con tarjeta de crédito en septiembre del 2013 por titulares de tarjetas europeas. Estas transacciones ocurrieron en 2 días, donde tenemos 492 fraudes de un total de 284807 transacciones.

> Con la ayuda de la ciencia de datos, buscaré determinar los clasificadores que me permitan tener un mayor clasificación.


In [10]:
import pandas as pd
df2= pd.read_csv("/content/drive/MyDrive/Primera Entrega (Data Science)/creditcard.csv")
df2.shape

(284807, 31)

Shape: Esta base de datos...

*   Tiene 284807 filas y 31 columnas.
*   Tiene 284807 registros y 31 variables.

In [11]:
df2.head()

Time        V1        V2        V3        V4        V5        V6        V7  \
0   0.0 -1.359807 -0.072781  2.536347  1.378155 -0.338321  0.462388  0.239599   
1   0.0  1.191857  0.266151  0.166480  0.448154  0.060018 -0.082361 -0.078803   
2   1.0 -1.358354 -1.340163  1.773209  0.379780 -0.503198  1.800499  0.791461   
3   1.0 -0.966272 -0.185226  1.792993 -0.863291 -0.010309  1.247203  0.237609   
4   2.0 -1.158233  0.877737  1.548718  0.403034 -0.407193  0.095921  0.592941   

         V8        V9  ...       V21       V22       V23       V24       V25  \
0  0.098698  0.363787  ... -0.018307  0.277838 -0.110474  0.066928  0.128539   
1  0.085102 -0.255425  ... -0.225775 -0.638672  0.101288 -0.339846  0.167170   
2  0.247676 -1.514654  ...  0.247998  0.771679  0.909412 -0.689281 -0.327642   
3  0.377436 -1.387024  ... -0.108300  0.005274 -0.190321 -1.175575  0.647376   
4 -0.270533  0.817739  ... -0.009431  0.798278 -0.137458  0.141267 -0.206010   

        V26       V27       V28  Amount  Class  
0 -0.189115  0.133558 -0.021053  149.62      0  
1  0.125895 -0.008983  0.014724    2.69      0  
2 -0.139097 -0.055353 -0.059752  378.66      0  
3 -0.221929  0.062723  0.061458  123.50      0  
4  0.502292  0.219422  0.215153   69.99      0  

[5 rows x 31 columns]

In [12]:
df2.columns

Index(['Time', 'V1', 'V2', 'V3', 'V4', 'V5', 'V6', 'V7', 'V8', 'V9', 'V10',
       'V11', 'V12', 'V13', 'V14', 'V15', 'V16', 'V17', 'V18', 'V19', 'V20',
       'V21', 'V22', 'V23', 'V24', 'V25', 'V26', 'V27', 'V28', 'Amount',
       'Class'],
      dtype='object')

##### *Descripción de las variables*

*   `Time`: Segundos transcurridos entre esta transacción y la primera transacción de los conjuntos de datos.
*   `V1-V28`: Estas variables estan trabajados bajo el metodo PCA para proteger la identidad de los usuarios.
*   `Amount`: Valor de la transacción
*   `Class`: 1 transacción fraudulenta, 0 transacción no fraudulenta.

#### ***2.3 Tercera base de datos***
#### Hotel Reservations Dataset
> Esta base de datos cuenta con información sobre las reservas del hotel realizadas en linea. Con algoritmos de machine learning se buscará encontrar patrones de comportamientos de los clientes para predecir si van a cancelar su reservación.

In [13]:
import pandas as pd
df3= pd.read_csv("/content/drive/MyDrive/Primera Entrega (Data Science)/Hotel Reservations.csv")
df3.shape

(36275, 19)

In [14]:
df3.head()

Booking_ID  no_of_adults  no_of_children  no_of_weekend_nights  \
0   INN00001             2               0                     1   
1   INN00002             2               0                     2   
2   INN00003             1               0                     2   
3   INN00004             2               0                     0   
4   INN00005             2               0                     1   

   no_of_week_nights type_of_meal_plan  required_car_parking_space  \
0                  2       Meal Plan 1                           0   
1                  3      Not Selected                           0   
2                  1       Meal Plan 1                           0   
3                  2       Meal Plan 1                           0   
4                  1      Not Selected                           0   

  room_type_reserved  lead_time  arrival_year  arrival_month  arrival_date  \
0        Room_Type 1        224          2017             10             2   
1        Room_Type 1          5          2018             11             6   
2        Room_Type 1          1          2018              2            28   
3        Room_Type 1        211          2018              5            20   
4        Room_Type 1         48          2018              4            11   

  market_segment_type  repeated_guest  no_of_previous_cancellations  \
0             Offline               0                             0   
1              Online               0                             0   
2              Online               0                             0   
3              Online               0                             0   
4              Online               0                             0   

   no_of_previous_bookings_not_canceled  avg_price_per_room  \
0                                     0               65.00   
1                                     0              106.68   
2                                     0               60.00   
3                                     0              100.00   
4                                     0               94.50   

   no_of_special_requests booking_status  
0                       0   Not_Canceled  
1                       1   Not_Canceled  
2                       0       Canceled  
3                       0       Canceled  
4                       0       Canceled

In [16]:
df3.columns

Index(['Booking_ID', 'no_of_adults', 'no_of_children', 'no_of_weekend_nights',
       'no_of_week_nights', 'type_of_meal_plan', 'required_car_parking_space',
       'room_type_reserved', 'lead_time', 'arrival_year', 'arrival_month',
       'arrival_date', 'market_segment_type', 'repeated_guest',
       'no_of_previous_cancellations', 'no_of_previous_bookings_not_canceled',
       'avg_price_per_room', 'no_of_special_requests', 'booking_status'],
      dtype='object')

##### *Descripción de las variables*

*   `Booking_ID`: Identificador de reserva.
*   `no_of_adults`: Número de adultos.
*   `no_of_children`: Número de niños.
*   `no_of_weekend_nights`: Número de noches de fin de semana que reservo.
*   `no_of_week_nights`: Número de noches de la semana que reservo.
*   `type_of_meal_plan`: Tipo de plan de comidas reservado.
*   `required_car_parking_space`: ¿Requiere espacio de estacionamiento? 0 No, 1 Sí
*   `room_type_reserved`: Tipo de habitación reservada.
*   `lead_time`: Número de días entre las fecha de reserva y fecha de llegada.
*   `arrival_year`: Año de la fecha de llegada.
*   `arrival_month`: Mes de la fecha de llegada.
*   `arrival_date`: Día de la fecha de llegada.
*   `market_segment_type`: Designación del segmento del mercado.
*   `repeated_guest`: ¿Es cliente recurrente? 0 No, 1 Sí
*   `no_of_previous_cancellations`: Número de reservas anteriores que fueron canceladas.
*   `no_of_previous_bookings_not_canceled`: Número de reservas anteriores que no fueron canceladas.
*   `avg_price_per_room`: Precio promedio por día de reserva.
*   `no_of_special_requests`: Número total de solicitudes especiales.
*   `booking_status`: Indicador de reservación.